### PR TITLE
Expose 'suix_getDynamicFieldObject' and 'suix_subscribeTransaction' through `ReadApi`

### PR DESCRIPTION
**Pull request by unmaykr-aftermath**
## Description 

Adds a way to call the 'suix_getDynamicFieldObject' and 'suix_subscribeTransaction' RPC methods through `SuiClient` via `client.read_api().get_dynamic_field_object(/* ... */)` and `client.read_api().subscribe_transaction(/* ... */)`.

## Test Plan 

N/A

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-sdk/src/apis.rs [hunk 1]
```diff
@@ -558,6 +573,23 @@ impl ReadApi {
         )
     }
 
+    /// Subscribe to a stream of Sui transaction effects
```
Comment (stefan-mysten) [inline crates/sui-sdk/src/apis.rs:576]:
```suggestion
    /// Subscribe to a stream of transactions. 
    ///
    /// This is only available through WebSockets. 
```

File: crates/sui-sdk/src/apis.rs [hunk 2]
```diff
@@ -138,6 +140,19 @@ impl ReadApi {
             .await?)
     }
 
+    /// Return the dynamic field object information for a specified object
```
Comment (stefan-mysten) [inline crates/sui-sdk/src/apis.rs:143]:
```suggestion
    /// Return the dynamic field object information for a specified object.
```

File: crates/sui-sdk/src/apis.rs [hunk 3]
```diff
@@ -558,6 +573,23 @@ impl ReadApi {
...
+        match &self.api.ws {
+            Some(c) => {
+                let subscription: Subscription<SuiTransactionBlockEffects> =
+                    c.subscribe_transaction(filter).await?;
+                Ok(subscription.map(|item| Ok(item?)))
+            }
+            _ => Err(Error::Subscription(
+                "Subscription only supported by WebSocket client.".to_string(),
+            )),
+        }
```
Comment (amnn) [inline crates/sui-sdk/src/apis.rs:590]: nit: preference for early return style here.

```suggestion
        let Some(c) = &self.api.ws else {
            return Err(Error::Subscription(
                "Subscription only supported by WebSocket client.".to_string(),
            ));
        };
        let subscription: Subscription<SuiTransactionBlockEffects> =
            c.subscribe_transaction(filter).await?;
        Ok(subscription.map(|item| Ok(item?)))
```

**Review by amnn — APPROVED** [review]
Thanks for contributing these @unmaykr-aftermath , they look good to me!

Could you also add some examples to in `crates/sui-sdk/examples/read_api.rs` that demonstrate how these endpoints work?  I'll leave it with @stefan-mysten to try those out and give the final stamp.

**Comment by unmaykr-aftermath** [comment]
> Thanks for contributing these @unmaykr-aftermath , they look good to me!
> 
> Could you also add some examples to in `crates/sui-sdk/examples/read_api.rs` that demonstrate how these endpoints work? I'll leave it with @stefan-mysten to try those out and give the final stamp.

Thanks @amnn! I added the examples as requested so that @stefan-mysten can give his feedback.

**Review by stefan-mysten — APPROVED** [review]
Great work @unmaykr-aftermath, thank you!

**Pull request merged**